### PR TITLE
Added support for source maps

### PR DESCRIPTION
**Pull request by soluml**
Added source map support for this plugin, also added documentation.

### CODE REVIEW

**Comment by gmarty** [comment]
Thanks Benjamin.
Can you explain why do you need a special treatment for generating source map? The setting in `options` are passed directly to Closure Compiler interns.
What didn't work with:

``` javascript
grunt.initConfig({
  'closure-compiler': {
    frontend: {
      closurePath: '/src/to/closure-compiler',
      js: 'static/src/frontend.js',
      options: {
        create_source_map: 'src.map',
        source_map_format: 'V3'
      }
    }
  }
});
```

**Comment by soluml** [comment]
Thanks for your response Guillaume, I totally missed that in the documentation. Creating the maps with the options object works great! Go ahead and ignore this request.

I've modified my code to only include the "sourceMapUrl" option, which appends the sourceMapUrl comment to the end of the generated JS file which associates the two together. I have no idea why this isn't built into Closure Compiler by default, but I wasn't able to find any documentation on a flag to turn this on. I'm working in a Jenkins environment where hand editing files is a no go and this way we don't have to manage a bunch of X-SourceMap headers on the files.

I'll add a new pull request for just this code.  Thanks!

**Comment by soluml** [comment]
Added a new pull request for just the sourceMapUrl feature: https://github.com/gmarty/grunt-closure-compiler/pull/43

Thanks again for pointing this out Guillaume!

**Comment by evenicoulddoit** [comment]
I'm using this along with the grunt watch plugin and it's adding the map to my root directory. IMO it should go into the same folder as the compiled JS.

File: tasks/closure-compiler.js [hunk 1]
```diff
@@ -94,6 +94,15 @@ module.exports = function(grunt) {
         done(false);
       }
 
+      // Check to see if we should add the source map comment to end of file
+      if (data.sourceMapUrl && data.options.create_source_map) {
+        if (typeof data.sourceMapUrl == 'boolean') data.sourceMapUrl = path.basename(data.jsOutputFile) + '.map';
```
Comment (evenicoulddoit) [inline tasks/closure-compiler.js:99]: This isn't checking for "truthy", you should use `if data.sourceMapUrl === true`

File: tasks/closure-compiler.js [hunk 2]
```diff
@@ -94,6 +94,15 @@ module.exports = function(grunt) {
         done(false);
       }
 
+      // Check to see if we should add the source map comment to end of file
+      if (data.sourceMapUrl && data.options.create_source_map) {
```
Comment (evenicoulddoit) [inline tasks/closure-compiler.js:98]: Where's the need for two options? Could you not just check for `data.options.create_source_map`?

**Comment by ChadKillingsworth** [comment]
> I have no idea why this isn't built into Closure Compiler by default

I'm a Closure-compiler project maintainer. You do not need any special options - you should use the `output_wrapper` option that already exists on the compiler to add the source map url comment. See https://github.com/google/closure-compiler/wiki/FAQ#source-maps-and-sourcemappingurl

**Pull request closed without merging**